### PR TITLE
Search queries report

### PR DESCRIPTION
**Pull request by JaabaShooler**
image
Add a menu component

### CODE REVIEW

File: pages/SearchQueries.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  React.useEffect(() => {
+    const onBoudyClick = (event: any) => {
+      if (ref && !ref.current!.contains(event.target)) {
+        setOpen(false);
+
+        return;
+      }
+    };
+    document.body.addEventListener('click', onBoudyClick);
```
Comment (hmelenok) [inline pages/SearchQueries.tsx:23]: Typo: `onBoudyClick` -> `onBodyClick`

File: src/components/Tools/Menu/Menu.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.menu {
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.scss:1]: Where is BEM?

File: src/components/Tools/Menu/Menu.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  background-color: white;
+  ul {
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.scss:3]: Please use classes

File: src/components/Tools/Menu/Menu.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            <li key={item.title} onClick={() => item.execute()}>
+              <img src="notNow" alt="exit " />
+              <span>{item.title}</span>
+            </li>
+          );
+        })}
+      </ul>
+    </div>
+  ) : (
+    <></>
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.tsx:24]: it can be null (as non existing element in React)

File: src/components/Tools/Menu/Menu.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+const Menu: React.FC<IMenuProps> = ({menuItemArr, open}) => {
+  return open ? (
+    <div className="menu">
+      <ul>
+        {menuItemArr.map(item => {
+          return (
+            <li key={item.title} onClick={() => item.execute()}>
+              <img src="notNow" alt="exit " />
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.tsx:16]: is it real image `src`?

File: src/components/Tools/Menu/Menu.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import './Menu.scss';
+
+interface IMenuProps {
+  menuItemArr: any[];
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.tsx:5]: Please describe type of menu item. As it typed no reason to put `Arr` to name

File: src/components/Tools/Menu/Menu.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    li {
+      list-style: none;
+      padding: 20px;
+      cursor: pointer;
+      display: flex;
+      justify-content: center;
+      position: relative;
+      img {
+        position: absolute;
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.scss:12]: Maybe there is better way?

**Review by hmelenok — CHANGES_REQUESTED** [review]
Please fix comments. And  please add screenshot with menu.

**Review by pihorb — COMMENTED** [review]
Better to use lower case for route names, because it's case sensitive. 
Now, open `searchqueries` will lead to error

File: pages/SearchQueries.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  const handleClickToOpen = () => {
+    setOpen(!open);
+  };
+  const ref = React.useRef<HTMLDivElement>(null);
+  React.useEffect(() => {
+    validateAuth();
+  }, []);
+
+  React.useEffect(() => {
+    const onBoudyClick = (event: any) => {
```
Comment (pihorb) [inline pages/SearchQueries.tsx:16]: please, avoid using `any`

File: src/components/Tools/Menu/Menu.tsx (hunk 5 above)
Comment (JaabaShooler) [reply src/components/Tools/Menu/Menu.tsx:16]: no

File: pages/SearchQueries.tsx (hunk 8 above)
Comment (JaabaShooler) [reply pages/SearchQueries.tsx:16]: I need some help to choose correct type

File: src/components/Tools/Menu/Menu.scss (hunk 2 above)
Comment (hmelenok) [reply src/components/Tools/Menu/Menu.scss:1]: @JaabaShooler Please name it from capital letter

File: src/components/Tools/Menu/Menu.scss [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        text-align: center;
+        display: block;
+        font-family: 'Open Sans', serif;
+        font-weight: 500;
+        font-size: 14px;
+        &:first-letter {
+          text-transform: uppercase;
+        }
+      }
+      img {
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.scss:25]: No direct `element` selectors

File: src/components/Tools/Menu/Menu.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+const Menu: React.FC<IMenuProps> = ({menuItemArr, open}) => {
+  return open ? (
+    <div className="menu">
+      <ul className="menu__nav">
+        {menuItemArr.map(item => {
+          return (
+            <li className="menu__nav__element" key={item.title} onClick={() => item.execute()}>
+              <Image
+                className="menu__nav__element"
```
Comment (hmelenok) [inline src/components/Tools/Menu/Menu.tsx:18]: why classes the same?

**Review by hmelenok — APPROVED** [review]

File: src/components/Tools/Menu/Menu.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import Image from 'next/image';
+import './Menu.scss';
+
+interface IMenuProps {
+  menuItemArr: [{title: string; execute: () => void}];
```
Comment (slavik-chapelskyi) [inline src/components/Tools/Menu/Menu.tsx:6]:
```suggestion
  items: [{title: string; onClick: () => void}];
```

**Pull request merged**